src/main/java/com/aws/greengrass/shadowmanager/sync/model/LocalUpdateSyncRequest.java

```diff
@@ -99,96 +99,83 @@ public void merge(LocalUpdateSyncRequest other) throws IOException {
     @Override
     public void execute(SyncContext context) throws SkipSyncRequestException, ConflictError,
             UnknownShadowException {
-        // synchronizing access to shadow sync information because it is accessed by multiple threads.
-        // In this method, shadow sync information is accessed within context of the sync thread.
-        // The other place is `isUpdateNecessary(SyncContext)`, which runs on the Nucleus'
-        // MqttClient event loop thread.
-        //
-        // Other types of sync requests don't need this lock because their implementation of
-        // `isUpdateNecessary(SyncContext)` does not access shadow sync information.
-        //
-        // The check for shadow sync information in this class' `isUpdateNecessary(SyncContext)`
-        // prevents unnecessary full syncs,
-        // (see https://github.com/aws-greengrass/aws-greengrass-shadow-manager/pull/106).
-        synchronized (context.getSynchronizeHelper().getThingShadowLock(this)) {
-            ShadowDocument shadowDocument;
-            try {
-                shadowDocument = new ShadowDocument(updateDocument);
-            } catch (IOException | InvalidRequestParametersException e) {
-                throw new SkipSyncRequestException(e);
-            }
-
-            SyncInformation currentSyncInformation = context.getDao()
-                    .getShadowSyncInformation(getThingName(), getShadowName())
-                    .orElseThrow(() -> new UnknownShadowException("Shadow not found in sync table"));
+        ShadowDocument shadowDocument;
```

Comment:
removed the `synchronized` block since not we're not reading the sync info from multiple threads anymore